docs/puzzles/pooling.md

```diff
@@ -0,0 +1,320 @@
+  ; p1 is key_value_list if escaping
+
+  ; pool_reward_amount is the value of the coin reward - this is passed in so that this puzzle will still work after halvenings
+  ; pool_reward_height is the block height that the reward was generated at. This is used to calculate the coin ID.
+  ; key_value_list is signed extra data that the wallet may want to publicly announce for syncing purposes
+
+  (include condition_codes.clib)
+  (include singleton_truths.clib)
+
+  ; takes a lisp tree and returns the hash of it
+  (defun sha256tree (TREE)
+      (if (l TREE)
+          (sha256 2 (sha256tree (f TREE)) (sha256tree (r TREE)))
+          (sha256 1 TREE)
+      )
+  )
+
+  (defun-inline calculate_pool_reward (pool_reward_height P2_SINGLETON_PUZZLE_HASH POOL_REWARD_PREFIX pool_reward_amount)
+    (sha256 (logior POOL_REWARD_PREFIX (logand (- (lsh (q . 1) (q . 128)) (q . 1)) pool_reward_height)) P2_SINGLETON_PUZZLE_HASH pool_reward_amount)
+  )
+
+  (defun absorb_pool_reward (POOL_PUZZLE_HASH my_inner_puzzle_hash my_amount pool_reward_amount pool_reward_id)
+    (list
+        (list CREATE_COIN my_inner_puzzle_hash my_amount)
+        (list CREATE_COIN POOL_PUZZLE_HASH pool_reward_amount)
+        (list CREATE_PUZZLE_ANNOUNCEMENT pool_reward_id)
+        (list ASSERT_COIN_ANNOUNCEMENT (sha256 pool_reward_id '$'))
+    )
+  )
+
+  (defun-inline travel_to_waitingroom (OWNER_PUBKEY WAITINGROOM_PUZHASH my_amount extra_data)
+    (list (list AGG_SIG_ME OWNER_PUBKEY (sha256tree extra_data))
+          (list CREATE_COIN WAITINGROOM_PUZHASH my_amount)
+    )
+  )
+
+  ; main
+
+  (if pool_reward_height
+    (absorb_pool_reward POOL_PUZZLE_HASH
+                        (my_inner_puzzle_hash_truth Truths)
+                        (my_amount_truth Truths)
+                        p1
+                        (calculate_pool_reward pool_reward_height P2_SINGLETON_PUZZLE_HASH POOL_REWARD_PREFIX p1)
+    )
+    (travel_to_waitingroom OWNER_PUBKEY WAITINGROOM_PUZHASH (my_amount_truth Truths) p1)
+  )
+)
+```
+
+As always, let's begin with the arguments:
+
+`POOL_PUZZLE_HASH` is the address to which we want to pay the pool when we win a block reward.  The pool uses this to verify that the singleton has been lent to them.
+
+`P2_SINGLETON_PUZZLE_HASH` is the puzzle hash that payments to this singleton will have.  This is the hash that farmers make their plots to.
+
+`OWNER_PUBKEY` is the public key that will sign the decision to leave the pool.  This is likely owned by the farmer.
+
+`POOL_REWARD_PREFIX` is a bit of a unique argument.  On the chia mainnet, this will always be `ccd5bb71183532bff220ba46c268991a00000000000000000000000000000000`.  It is the beginning of the `parent_coin_info` that is assigned to the pool rewards since they do not have a parent.  That first half is followed by 32 zeroes to make it the length of a standard hash.  We'll talk more about why this is needed when we go over the segment of code that uses it.
```

Comment:
```suggestion
`POOL_REWARD_PREFIX` is a bit of a unique argument.  On the chia mainnet, this will always be `ccd5bb71183532bff220ba46c268991a00000000000000000000000000000000`.  It is the beginning of the `parent_coin_id` of pool reward coins. Because those coins do not have a parent, their ID is derived from a combination of the network ID in the left half of the ID, and an incrementing identifier in the right.  That first half is followed by 32 zeroes to make it the length of a standard hash.  We'll talk more about why this is needed when we go over the segment of code that uses it.
```